cirq-google/cirq_google/serialization/circuit_serializer.py

```diff
@@ -485,54 +489,52 @@ def _deserialize_circuit(
         deserialized_constants: List[Any],
     ) -> cirq.Circuit:
         moments = []
-        constant_moments: dict[int, cirq.Moment] = {}
-        constant_ops: dict[int, cirq.Operation] = {}
         for moment_proto in circuit_proto.moments:
```

Comment:
Can we add a validator here that `moments` and `moment_indices` are exclusive?
```python
if circuit_proto.moments and circuit_proto.moment_indices:
    raise ValueError('Circuit message must not have "moments" and "moment_indices" fields set at the same time.') 
```

Reply:
Done and added test.